testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeDatabricksUnityCompatibility.java

```diff
@@ -0,0 +1,224 @@
+                    .containsOnly(expectedRowsForInsert);
+            assertThat(onDelta().executeQuery(format("SELECT * FROM %s.%s.%s order by c1", unityCatalogCatalogName, schemaName, tableName)))
+                    .containsOnly(expectedRowsForInsert);
+
+            // update
+            List<QueryAssert.Row> expectedRowsForUpdate = ImmutableList.of(row(1, "one"), row(2, "two hundred"));
+            onTrino().executeQuery(format("UPDATE delta.%s.%s SET c2 = 'two hundred' WHERE c1 = 2", schemaName, tableName));
+            assertThat(onTrino().executeQuery(format("SELECT * FROM delta.%s.%s order by c1", schemaName, tableName)))
+                    .containsOnly(expectedRowsForUpdate);
+            assertThat(onDelta().executeQuery(format("SELECT * FROM %s.%s.%s order by c1", unityCatalogCatalogName, schemaName, tableName)))
+                    .containsOnly(expectedRowsForUpdate);
+
+            // delete
+            List<QueryAssert.Row> expectedRowsForDelete = ImmutableList.of(row(2, "two hundred"));
+            onTrino().executeQuery(format("DELETE FROM delta.%s.%s WHERE c2 = 'one'", schemaName, tableName));
+            assertThat(onTrino().executeQuery(format("SELECT * FROM delta.%s.%s order by c1", schemaName, tableName)))
+                    .containsOnly(expectedRowsForDelete);
+            assertThat(onDelta().executeQuery(format("SELECT * FROM %s.%s.%s order by c1", unityCatalogCatalogName, schemaName, tableName)))
+                    .containsOnly(expectedRowsForDelete);
+
+            // merge
+            List<QueryAssert.Row> expectedRowsForMerge = ImmutableList.of(row(1, "one"), row(2, "two"), row(3, "three"));
+            String sourceTableName = "test_dt_source_" + randomNameSuffix();
+            String tableLocation2 = getLocationForTable(externalLocationPath, schemaName, sourceTableName);
+            onDelta().executeQuery(
+                    format("CREATE TABLE %s.%s.%s (c1 int, c2 string) using delta location '%s'",
+                            unityCatalogCatalogName, schemaName, sourceTableName, tableLocation2));
+            onDelta().executeQuery(
+                    format("INSERT INTO %s.%s.%s values (1, 'one'), (2, 'two'), (3, 'three')",
+                            unityCatalogCatalogName, schemaName, sourceTableName));
+
+            onTrino().executeQuery(format("MERGE INTO delta.%s.%s t USING delta.%s.%s s on t.c1 = s.c1 " +
+                    "WHEN MATCHED THEN UPDATE SET c2 = s.c2 " +
+                    "WHEN NOT MATCHED THEN INSERT (c1, c2) VALUES (s.c1, s.c2)", schemaName, tableName, schemaName, sourceTableName));
+            assertThat(onTrino().executeQuery(format("SELECT * FROM delta.%s.%s order by c1", schemaName, tableName)))
+                    .containsOnly(expectedRowsForMerge);
+            assertThat(onDelta().executeQuery(format("SELECT * FROM %s.%s.%s order by c1", unityCatalogCatalogName, schemaName, tableName)))
+                    .containsOnly(expectedRowsForMerge);
```

Comment:
`containsOnly` doesn't verify the order. Please remove `order by` from SELECT statement.